charms/benchmark/__init__.py

```diff
@@ -118,9 +120,26 @@ def start():
         If the cabs-collector charm is installed, take a snapshot
         of the current profile data.
         """
+        # Do profile data collection immediately on this unit
         if os.path.exists(COLLECT_PROFILE_DATA):
             subprocess.check_output([COLLECT_PROFILE_DATA])
 
+        # Tell the benchmark-gui charm the action_id via the benchmark
+        # relation. Benchmark-gui will pass the action_id to all
+        # collector charms in the environment (via the collector relation),
+        # triggering profile data collection on each.
+        if os.environ['CHARM_DIR']:
```

Comment:
Someone could call benchmark-start from the CLI, which would throw an error otherwise. It might be good to throw an else in there to inform the user that it must be run from a hook context.

Also, I think we shouldn't collect profile data unless we're in a hook context.